cli/src/base.ts

```diff
@@ -5,16 +5,44 @@ import chalk from 'chalk';
 import dotenv from 'dotenv';
 import fetch from 'node-fetch';
 import table from 'text-table';
+import { z } from 'zod';
 import { readAPIKey } from './key.js';
+import { cosmiconfigSync } from 'cosmiconfig';
+
+export const userConfig = z.object({
+  codegen: z.optional(
+    z.object({
+      output: z.string()
+    })
+  )
+});
+
+export type UserConfig = z.infer<typeof userConfig>;
 
 export abstract class BaseCommand extends Command {
   // Date formatting is not consistent across locales and timezones, so we need to set the locale and timezone for unit tests.
   // By default this will use the system locale and timezone.
   locale: string | undefined = undefined;
   timeZone: string | undefined = undefined;
 
+  userConfig: UserConfig | undefined;
+
   async init() {
     dotenv.config();
+
+    const search = cosmiconfigSync('xata').search();
+    if (search) {
```

Comment:
Added an item here https://github.com/xataio/client-ts/issues/237 to print the filepath found with a _verbose_ option.